doc/designs/ldap_pam_passthrough.md

```diff
@@ -0,0 +1,327 @@
+# LDAP PAM Passthrough support
+
+## Overview
+
+Many organizations have authentication mechanisms already in place.
+They may not want to have IPA be the central repository for authentication.
+
+RADIUS is a common authentication protocol used for external authentication
+into existing systems. IPA currently has support for verifying credentails
+over RADIUS for Kerberos connections using the radius authentication
+indicator, but this does not work with LDAP authentication.
+
+For this document "PAM Passthrough" is defined as any subsequent
+plugin that handles authentication of the user entry using the PAM stack.
+
+There are two known possibilities:
+
+- The 389-ds PAM Passthrough Plugin
+- The slapi-nis 389-ds plugin
```

Comment:
Sorry for making a confusion here. I did not suggest to specify `slapi-nis` as one of authentication plugins here. What I suggested is that we could incorporate the code that we have in `slapi-nis` into `ipa-pwd-extop` so that it would cover PAM authentication on itself, without failing over to other authentication plugins.

`slapi-nis` plugin only provides PAM authentication for its own Compatibility Tree entries that correspond to AD users, so by design it would not be able to handle DNs in the primary tree.

Reply:
Ok I can clarify something like the way slapi-nis does it, but it basically uses a copy of the 389-ds pam pass through plugin...

I was trying to make this backend agnostic. I don't think it really matters design-wise what does the PAM auth, as long as it succeeds and fails when it should, and we do the right thing in that case.